make test doesn't work

`make test` no longer works.  It gives the following error on my machine:

```
python3 tests.py
Traceback (most recent call last):
  File "/home/daw/datascience/datascience/tests.py", line 5, in <module>
    sys.exit(pytest.main(['--nbval-lax', '--cov=datascience', 'tests']))
  File "/home/daw/datascience/lib64/python3.10/site-packages/_pytest/config/__init__.py", line 136, in main
    config = _prepareconfig(args, plugins)
  File "/home/daw/datascience/lib64/python3.10/site-packages/_pytest/config/__init__.py", line 313, in _prepareconfig
    config = pluginmanager.hook.pytest_cmdline_parse(
  File "/home/daw/datascience/lib64/python3.10/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/daw/datascience/lib64/python3.10/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/daw/datascience/lib64/python3.10/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/home/daw/datascience/lib64/python3.10/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/home/daw/datascience/lib64/python3.10/site-packages/_pytest/helpconfig.py", line 99, in pytest_cmdline_parse
    config = outcome.get_result()  # type: Config
  File "/home/daw/datascience/lib64/python3.10/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/daw/datascience/lib64/python3.10/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/daw/datascience/lib64/python3.10/site-packages/_pytest/config/__init__.py", line 932, in pytest_cmdline_parse
    self.parse(args)
  File "/home/daw/datascience/lib64/python3.10/site-packages/_pytest/config/__init__.py", line 1204, in parse
    self._preparse(args, addopts=addopts)
  File "/home/daw/datascience/lib64/python3.10/site-packages/_pytest/config/__init__.py", line 1097, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest11")
  File "/home/daw/datascience/lib64/python3.10/site-packages/pluggy/manager.py", line 299, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/usr/lib64/python3.10/importlib/metadata/__init__.py", line 171, in load
    module = import_module(match.group('module'))
  File "/usr/lib64/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "/home/daw/datascience/lib64/python3.10/site-packages/_pytest/assertion/rewrite.py", line 161, in exec_module
    source_stat, co = _rewrite_test(fn, self.config)
  File "/home/daw/datascience/lib64/python3.10/site-packages/_pytest/assertion/rewrite.py", line 360, in _rewrite_test
    co = compile(tree, fn_, "exec", dont_inherit=True)
TypeError: required field "lineno" missing from alias
make: *** [Makefile:20: test] Error 1
```

I suspect the issue is the use of pytest 6.0.1.  It sounds like changes to the `ast` package broke pytest: https://bugs.python.org/issue43798, https://github.com/pytest-dev/pytest/issues/8539.  And, `requirements-testing.txt` currently pins pytest to version 6.0.1.

Updating to pytest 6.2.5 fixes this specific problem (though now it surfaces a new issue).  So, a plausible fix is to replace `pytest==6.0.1` with `pytest` (or `pytest<7`) in `requirements-testing.txt`.  Is that the right fix?

Is there any reason to pin to version 6 of pytest, or can we update it to use the latest version of pytest (pytest 7)?

--- a/.github/workflows/run_tests.yml
+++ b/.github/workflows/run_tests.yml
@@ -16,7 +16,7 @@ jobs:
       with:
         activate-environment: datascience
         environment-file: linux_environment.yml
-        python-version: 3.6
+        python-version: 3.10
         auto-activate-base: true
 
     - name: Add conda to system path

--- a/.github/workflows/run_tests.yml
+++ b/.github/workflows/run_tests.yml
@@ -16,7 +16,7 @@ jobs:
       with:
         activate-environment: datascience
         environment-file: linux_environment.yml
-        python-version: 3.10
+        python-version: '3.10'
         auto-activate-base: true
 
     - name: Add conda to system path

--- a/datascience/tables.py
+++ b/datascience/tables.py
@@ -4627,7 +4627,7 @@ class Table(collections.abc.MutableMapping):
             type(self).plots.append(axis)
         else:
             fig, axes = plt.subplots(n, 1, figsize=(width, height*n))
-            if not isinstance(axes, collections.Iterable):
+            if not isinstance(axes, collections.abc.Iterable):
                 axes=[axes]
             for axis, y_label, color in zip(axes, y_labels, colors):
                 draw(axis, y_label, color)

--- a/datascience/tables.py
+++ b/datascience/tables.py
@@ -2877,13 +2877,14 @@ class Table(collections.abc.MutableMapping):
                 uvw  | 20   | 5.9
 
                 >>> table_as_html = table.as_html()
-                <table border="1" class="dataframe">\\n    <thead>\\n        <tr>\\n            
+                >>> table_as_html
+                '<table border="1" class="dataframe">\\n    <thead>\\n        <tr>\\n            
                 <th>name</th> <th>age</th> <th>height</th>\\n        
                 </tr>\\n    </thead>\\n    <tbody>\\n        
                 <tr>\\n            <td>abc </td> <td>12  </td> <td>5.5   </td>\\n        </tr>\\n        
                 <tr>\\n            <td>xyz </td> <td>14  </td> <td>6     </td>\\n        </tr>\\n        
                 <tr>\\n            <td>uvw </td> <td>20  </td> <td>5.9   </td>\\n        </tr>\\n    
-                </tbody>\\n</table>
+                </tbody>\\n</table>'
 
                 2. Simple table being converted to HTML with max_rows passed in
 

--- a/requirements-tests.txt
+++ b/requirements-tests.txt
@@ -1,4 +1,4 @@
-pytest==6.0.1
+pytest==7.1.2
 coverage
 coveralls
 bokeh

